### PR TITLE
feat(youtube): support for 18.23.35

### LINKED ISSUE #895: bug(youtube/old-quality-layout): old quality menu enabled, but still shows new quality menu
### Type

Error at runtime

### Bug description

Enabling the old quality menu in settings still shows the new quality menu.

### Steps to reproduce

- Patch with old-quality-layout
- The new quality layout is still shown

### Relevant log output

```shell
N/A
```

### Screenshots or videos

Screenshot_20230525-091503561
Screenshot_20230525-091525689

### Solution

_No response_

### Additional context

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

### PR DESCRIPTION
**Pull request by LisoUseInAIKyrios**
Changes to support YouTube 18.23.35

#### Issues to resolve:

- [x] https://github.com/revanced/revanced-patches/issues/2307
- [x] https://github.com/revanced/revanced-patches/issues/2250
- [x] `ProtobufBufferFingerprint` needs updating
- [x] RYD fingerprint needs updating
- [x] minimized-playback fingerprint needs updating
- [x] Deduplicate `onFlyoutCreate` methods
- [x] Resolve a small TODO in `SettingsEnum` (Disable custom speeds)
- [x] Remember video quality in new menu
- [x] Remember video speed in new menu (old works)

## Depends on 

- https://github.com/revanced/revanced-integrations/pull/433

### CODE REVIEW

**Comment by oSumAtrIX** [comment]
Has the patch that restores the old quality menu layout been tested? Iirc that one was something that was changed

**Comment by oSumAtrIX** [comment]
> but old quality menu does not work

This means the event handler has been changed. It is likely something related to Litho and I am not sure of Litho handles that.

**Comment by oSumAtrIX** [comment]
> but old quality menu does not work
> ...
> Custom playback speed is also nonfunctional

I'd say this shouldn't block this PR. Remove the compatibility to the new version for these patches and let's proceed.

**Comment by oSumAtrIX** [comment]
If it turns out to be difficult to support the older versions, you can drop compatibility for it.

**Comment by oSumAtrIX** [comment]
I am working on fixing the remaining issues and will push shortly.

**Comment by 0xrxL** [comment]
> There are no apparent changes between the latest YouTube and 18.19.35.
> 
> Considering the issues with the new speed and quality menus that now use litho components, and the possibility for new ads or new A/B tests in the latest app version, for now the easy answer is don't upgrade the app target version until there's a benefit to do so.

Both speed and quality menus has been fixed by @inotia00 

I already have the fix for quality menu (only this, for personal use) on my repo. If you're interested to port it I can give you the access to it. 👍

**Comment by oSumAtrIX** [comment]
I have a local prototype ready already.

**Comment by 0xrxL** [comment]
> I have a local prototype ready already.

Oh...good so.

**Comment by inotia00** [comment]
Also, `Remember video quality changes` and `Remember playback speed changes` do not work in the new player flyout panel.

Reversing in an abstract class was quite tricky, but I finally figured it out

Please refer to the following two commits:

fix: `Remember video quality changes`
https://github.com/inotia00/revanced-patches/commit/0d04270d667cbcac2982d811325a1e61922cf604

fix: `Remember playback speed changes`
https://github.com/inotia00/revanced-patches/commit/70761a667671d6b9ab462f323415e4bce8854bb4

I didn't add comments to this commit, so if you have any questions feel free to ask.

**Comment by oSumAtrIX** [comment]
The current implementation of restoring the old quality menu is intervined with the patch to show custom playback speeds. In order to make the custom playback speeds patch work separately, an onClick listener has to be added to the Advanced qualities menu item, so that it opens the old quality menu instead of the new one. This would allow seperating the patch from the old quality menu patch

**Comment by inotia00** [comment]
yeah.. the observer is added to the same view, so the patch won't be separate as you say

I temporarily circumvented this issue in a rather inefficient way - by checking whether the integrations contained the patch or not

[PatchStatus.OldQualityLayout()](https://github.com/inotia00/revanced-integrations/blob/263f51de03c17b5bff7da0d6774f7bb5cf046895/app/src/main/java/app/revanced/integrations/patches/ads/LowLevelFilter.java#L85)
[PatchStatus.OldSpeedLayout()](https://github.com/inotia00/revanced-integrations/blob/263f51de03c17b5bff7da0d6774f7bb5cf046895/app/src/main/java/app/revanced/integrations/patches/ads/LowLevelFilter.java#L90)


Since this is not the ideal way, so I will have to find an alternative as you said

**Comment by oSumAtrIX** [comment]
@LisoUseInAIKyrios Can you look into resolving the small TODO? In case it is easily possible, also deduplicate the ´onFlyoutCreate` methods, there's two, but they aren't too similar, so if it turns out to worsen readability, mark as checked.

**Comment by oSumAtrIX** [comment]
Yes. Assume the following scenarios:

I include the patch, but I don't want the custom speeds to appear anymore. Keep in mind that this patch is included by default. The only way to get rid of them would be either repatching or having to manually remove all the custom speeds whereas options at runtime is all about avoiding to repatch. 

Another scenario is when you do not want the old "legacy" menu to appear and instead the new menu. The new menu is slightly different and has a "Normal" entry. Though theres no functional benefit in this one.

If you don't consider both scenarios to be sufficient as a motivation for the option, you can remove it

File: src/main/kotlin/app/revanced/patches/youtube/video/speed/custom/patch/CustomVideoSpeedPatch.kt [hunk 1]
```diff
@@ -54,15 +54,15 @@ class CustomVideoSpeedPatch : BytecodePatch(
...
                         ),
                         summaryOn = StringResource(
                             "revanced_custom_playback_speeds_enabled_summary",
-                            "Custom playback speeds are visible"
+                            "Using custom playback speeds"
                         ),
                         summaryOff = StringResource(
                             "revanced_custom_playback_speeds_enabled_summary",
-                            "Default playback speeds are visible"
+                            "Using YouTube default playback speeds "
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/video/speed/custom/patch/CustomVideoSpeedPatch.kt]: Probably unnecessary to have "YouTube" here.

File: src/main/kotlin/app/revanced/patches/youtube/video/quality/fingerprints/NewVideoQualityChangedFingerprint.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        Opcode.MOVE_RESULT_OBJECT,
+        Opcode.IF_NEZ,
+        Opcode.SGET_OBJECT,
+        Opcode.GOTO,
+        Opcode.SGET_OBJECT,
+        Opcode.INVOKE_VIRTUAL,
+        Opcode.SGET_OBJECT,
+        Opcode.RETURN_OBJECT,
+    )
+)
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/video/quality/fingerprints/NewVideoQualityChangedFingerprint.kt]: Quite a big fingerprint; the pattern may be reduced

**Review by oSumAtrIX — APPROVED** [review]

**Comment by oSumAtrIX** [comment]
Odd, in my tests I was able to make it save any resolution I picked

**Pull request merged**